commit c16f3f75bc9668381a2426bc5e8a2c8739fa0c29
Author: Mark Robert Miller <markrmiller@apache.org>
Date:   Sun Jul 15 21:30:13 2012 +0000

    SOLR-3620: check if shutdown after wait so that any threads that were waiting bail rather than starting a recovery
    
    git-svn-id: https://svn.apache.org/repos/asf/lucene/dev/trunk@1361805 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/solr/core/src/java/org/apache/solr/update/DefaultSolrCoreState.java b/solr/core/src/java/org/apache/solr/update/DefaultSolrCoreState.java
index c8ba864..bc9f3ec 100644
--- a/solr/core/src/java/org/apache/solr/update/DefaultSolrCoreState.java
+++ b/solr/core/src/java/org/apache/solr/update/DefaultSolrCoreState.java
@@ -138,6 +138,11 @@ public final class DefaultSolrCoreState extends SolrCoreState {
         } catch (InterruptedException e) {
 
         }
+        // check again for those that were waiting
+        if (cc.isShutDown()) {
+          log.warn("Skipping recovery because Solr is shutdown");
+          return;
+        }
         if (closed) return;
       }
 
